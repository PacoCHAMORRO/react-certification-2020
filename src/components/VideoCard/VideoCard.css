.card {
  width: 25%;
  border-radius: 15px;
  box-shadow: 0px 3px 3px 3px #ccc;
  margin: 1rem;
  padding: 0.7rem;
}

.thumbnail-container {
  overflow: hidden;
  position: relative;
  border-radius: 15px;
  height: 240px;
  width: 100%;
}

.thumbnail {
  position: absolute;
  width: auto;
  height: 240px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  border-radius: 15px;
  margin: 0;
  z-index: -10;
}

.card-title {
  margin: 0;
  font-family: 'Montserrat', sans-serif;
  font-weight: 600;
}

.card-content p {
  font-weight: 200;
  font-size: 0.8rem;
  color: #333;
}

@media (max-width: 1124px) {
  .card {
    width: 40%;
  }
}

@media (max-width: 520px) {
  .card {
    width: 90%;
  }
}
