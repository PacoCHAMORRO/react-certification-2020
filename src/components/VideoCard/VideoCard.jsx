import React from 'react';

import './VideoCard.css';

const VideoCard = ({ thumbnail, title, description }) => (
  <div className="card">
    <div className="thumbnail-container">
      <img className="thumbnail" src={thumbnail} alt="Video snippet" />
    </div>
    <div className="card-body">
      <h3 className="card-title">{title}</h3>
      <div className="card-content">
        <p>{description}</p>
      </div>
    </div>
  </div>
);

export default VideoCard;
