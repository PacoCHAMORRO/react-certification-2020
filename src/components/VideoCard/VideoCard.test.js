import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import VideoCard from './VideoCard';

it('renders card component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<VideoCard />, div);
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <VideoCard
        thumbnail="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
        title="This is the video title"
        description="This ist hen the video despcription"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
