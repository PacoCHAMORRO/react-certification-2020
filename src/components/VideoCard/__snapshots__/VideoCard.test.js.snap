// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <img
    alt=""
    className="thumbnail"
    src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s800-c-k-c0xffffffff-no-rj-mo"
  />
  <div
    className="card-body"
  >
    <h3
      className="card-title"
    >
      This is the video title
    </h3>
    <div
      className="card-content"
    >
      This ist hen the video despcription
    </div>
  </div>
</div>
`;
